### ISSUE TITLE
AWS backend: WorkflowDockerLookupActor complains about AWS ECR registry

### ISSUE THREAD

**Issue by delocalizer** [broadinstitute/cromwell#4171]
The following Exception occurs using images from AWS ECR registry with the AWS backend; it appears harmless (the workflow proceeds to run just fine on AWS Batch) but is annoying in the logs:
[2018-09-18 12:14:53,68] [warn] BackendPreparationActor_for_27e08249:wf_globtest.globtest:-1:1 [27e08249]: Docker lookup failed
java.lang.Exception: Registry Some(368678863429.dkr.ecr.ap-southeast-2.amazonaws.com) is not supported
        at cromwell.engine.workflow.WorkflowDockerLookupActor.cromwell$engine$workflow$WorkflowDockerLookupActor$$handleLookupFailure(WorkflowDockerLookupActor.scala:196)

**Comment by TMiguelT** [comment]
+1 to this. I'm getting the same warning, which looks like a failure but actually seems to work fine.

**Comment by Horneth** [comment]
I think the real underlying issue is that we don't support retrieving docker hashes from ECR yet. See this issue: https://github.com/broadinstitute/cromwell/issues/3822
In the meantime you could either [disable fetching of the hash](https://github.com/broadinstitute/cromwell/blob/fdc6fdfd2381e08c5fbf84e171eaba30ef872beb/core/src/main/resources/reference.conf#L326) (**which will disable call caching for ALL dockers with a tag**) or try using a hash in your WDL instead of a floating tag.
Note that if you use a tag and Cromwell can't fetch the corresponding hash, call caching will be disabled for that task.

**Comment by alartin** [comment]
Actually it is not the only AWS ECR issue, it looks like any private repository has same issue.

**Comment by TMiguelT** [comment]
Yes, it even happens with Google Cloud repos, interestingly

**Comment by gf-atebbe** [comment]
+1 from AWS ECR

**Comment by geoffjentry** [comment]
There's a WIP PR #4864 to address this. We're not going to be working on it in the near future but it's likely not far from being ready if someone would like to take a stab at finishing it off

**Comment by microbioticajon** [comment]
Hi All, just checking in on this issue, to see if it is still alive. Would love to see ecr supported for hash-lookup.

**Comment by aednichols** [comment]
Not really, no. AWS hasn't been worked on with any substance [since 2021](https://github.com/broadinstitute/cromwell/pulls?q=is%3Apr+is%3Amerged+aws).
